### PR TITLE
Feature/menu refactor

### PR DESCRIPTION
**Pull request by dgibson666**
* Renamed the component and CSS classes to "menu" - the pattern is "overflow menu".
* Rewrote the menu functionality completely
* Added menu positioning and responsive behavior
* Added story controls

There is still some commented out CSS that I'd like to discuss before this gets merged.

### CODE REVIEW

File: packages/vanilla/src/utilities/createUniqueId.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+const createUniqueID = (prefix) =>
+  (prefix ? prefix + '-' : '') + Math.random().toString(36).slice(2, 7);
+
+export { createUniqueID };
```
Comment (dannyk3941) [inline packages/vanilla/src/utilities/createUniqueId.js:4]: You can write this as `export const createUniqueID = ....` and remove line 4.

**Review by dannyk3941 — CHANGES_REQUESTED** [review]
- One issue I'm seeing is that when the menu control/button is focused, hovering the button has a different background color than the border.
- Tabbing out of the menu when it is open does not close the menu.

File: packages/vanilla/src/index.js [hunk 2]
```diff
@@ -8,15 +9,18 @@ import Dropdown from './components/dropdown/dropdown';
...
 import SegmentedButtonGroup from './components/segmented-button-group/segmented-button-group';
 import Tabset from './components/tabs/tabs';
 import IndeterminateCheckbox from './components/checkbox/indeterminate';
 //import UniversalHeader from './components/header/header';
+
 import DarkMode from './utilities/darkMode';
 
-import './sass/main.scss';
+
+//import './sass/main.scss';
```
[the remark this answers is not in the document]
Comment (dgibson666) [reply packages/vanilla/src/index.js:23]: Good question. I was trying to figure out why we were importing SASS into the JS, but this makes sense I guess.

File: packages/vanilla/src/sass/components/_fileinput.scss [hunk 3]
```diff
@@ -64,8 +64,8 @@
   }
 
   &:focus-within {
-    border-color: var(--cbp-color-interactive-focus-darker);
-    outline-color: var(--cbp-color-interactive-focus-dark);
+    border-color: var(--cbp-color-interactive-focus-darker); // TechDebt: not to spec
```
[the remark this answers is not in the document]
Comment (dgibson666) [reply packages/vanilla/src/sass/components/_fileinput.scss:67]: I'm fixing this in this branch, but the focus colors of the selected file(s) remove buttons are also off. I'm fixing those in the separate bugfix branch.

File: packages/vanilla/src/sass/components/_menu.scss [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      font-weight: var(--cbp-font-weight-bold);
+      text-transform: uppercase;
+      color: var(--cbp-color-text-lightest);
+      letter-spacing: var(--cbp-letter-spacing-loose);
+      background-color: var(--cbp-color-interactive-primary-base);
+      border: 0;
+      cursor: pointer;
+
+      i {
+        margin-right: 8px;
```
[the remark this answers is not in the document]
Comment (dgibson666) [reply packages/vanilla/src/sass/components/_menu.scss:122]: done

**Pull request merged**